autogen/agentchat/groupchat.py

```diff
@@ -385,9 +396,40 @@ def _prepare_and_select_agents(
             ]
         return selected_agent, graph_eligible_agents, select_speaker_messages
 
+    def _call_custom_speaker_selection_func(self, last_speaker: Agent) -> None:
```

Comment:
The code organization seems overly complicated. Why don't you just put the logic in `_call_custom_speaker_selection_func` to `_prepare_and_select_agents` and remove `_call_custom_speaker_selection_func`. It seems not necessary.

Reply:
That does simply the code, thanks!